static/app/views/starfish/queries/useFullSpanDescription.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEventJSON} from 'sentry/views/starfish/queries/useEventJSON';
+import {useIndexedSpans} from 'sentry/views/starfish/queries/useIndexedSpans';
+import {SpanIndexedFields} from 'sentry/views/starfish/types';
+
+// NOTE: Fetching the top one is a bit naive, but works for now. A better
+// approach might be to fetch several at a time, and let the hook consumer
+// decide how to display them
+export function useFullSpanDescription(group: string) {
```

Comment:
l: Can this just return `string | undefined` for now then instead of an object? Usage of the function can just be:

```
const fullSpanDescription = useFullSpanDescription(groupId) ?? '';
```

This should be fairly easy to refactor afterwards as well.

If we do it this way, we can early return on creation of `firstIndexedSpan`, and then not have to deal with the `firstIndexedSpan` ternary checks.

Reply:
We could, but I don't like to hide when a hook makes a server request! `useFullSpanDescription` also returns properties like `isLoading` which is not important in _this_ UI, but will be important in UIs I'm starting work on. Plus, see note above about aliasing data